applications/MeshMovingApplication/custom_processes/impose_mesh_motion_process.cpp

```diff
@@ -0,0 +1,236 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+
+// Internal inlcudes
+#include "impose_mesh_motion_process.h"
+
+// Project includes
+#include "includes/checks.h"
+#include "utilities/parallel_utilities.h"
+#include "includes/mesh_moving_variables.h"
+
+
+namespace Kratos
+{
+
+
+ImposeMeshMotionProcess::ImposeMeshMotionProcess(ModelPart& rModelPart, Parameters parameters) :
+    Process(),
+    mrModelPart(rModelPart)
+{
+    KRATOS_TRY;
+
+    this->LoadFromParameters(parameters);
+
+    KRATOS_CATCH("");
+}
+
+
+void ImposeMeshMotionProcess::LoadFromParameters(Parameters parameters)
+{
+    KRATOS_TRY;
+
+    parameters.ValidateAndAssignDefaults(this->GetDefaultParameters());
+    Vector vector_parameter;
+
+    // Parse interval
+    vector_parameter = parameters.GetValue("interval").GetVector();
+    KRATOS_CHECK(vector_parameter.size() == 2);
+    mInterval[0] = vector_parameter[0];
+    mInterval[1] = vector_parameter[1];
+
+    // Parse translation
+    array_1d<double,3> translation_vector;
+    vector_parameter = parameters.GetValue("translation_vector").GetVector();
+    KRATOS_CHECK(vector_parameter.size() == 3);
```

Comment:
We usually give a meaningful error message, e.g. 
`KRATOS_ERROR << "translation_vector must have a size of 3" << std::endl;`

This is much nicer from a user POV

=> same for other checks